src/net/inv.rs

```diff
@@ -0,0 +1,1509 @@
+                                    test_debug!("Peer {:?} is diverged", nk);
+                                    self.diverged_peers.insert(nk);
+                                },
+                                NackResult::Broken => {
+                                    test_debug!("Peer {:?} is broken", nk);
+                                    self.broken_peers.insert(nk);
+                                }
+                            };
+                        },
+                        _ => {
+                            // unexpected reply
+                            test_debug!("Remote neighbor {:?} sent an unexpected reply of '{}'", &nk, message.get_message_name());
+                            self.broken_peers.insert(nk);
+                        }
+                    }
+
+                    None
+                },
+                Err(req_res) => {
+                    match req_res {
+                        Ok(same_req) => {
+                            // try again 
+                            Some(same_req)
+                        },
+                        Err(_e) => {
+                            // connection broken.
+                            // Don't try to contact this node again.
+                            test_debug!("Failed to get block inventory from {:?}: {:?}", &nk, &_e);
+                            self.dead_peers.insert(nk);
+                            None
+                        }
+                    }
+                }
+            };
+
+            if let Some(rh) = new_rh {
+                // still working
+                pending_getblocksinv_requests.insert(rh_nk, rh);
+            }
+        }
+
+        test_debug!("Still waiting for {} blocksinv replies", pending_getblocksinv_requests.len());
+
+        // are we done?
+        if pending_getblocksinv_requests.len() == 0 {
+            // don't call this again!
+            self.state = InvWorkState::Done;
+            return Ok(true);
+        }
+
+        // still have more to go 
+        for (nk, rh) in pending_getblocksinv_requests.drain() {
+            self.getblocksinv_requests.insert(nk, rh);
+        }
+        return Ok(false);
+    }
+}    
+
+impl PeerNetwork {
+    /// Given a sortition height of a block we're interested in, make a GetBlocksInv whose
+    /// _lowest_ block will be at the target sortition height.
+    fn make_highest_getblocksdata(&self, nk: &NeighborKey, target_block_height: u64, burndb: &mut BurnDB) -> Result<Option<GetBlocksInv>, net_error> {
+        let (mut highest_block_height, mut num_blocks) = 
+            if target_block_height + (BLOCKS_INV_DATA_MAX_BITLEN as u64) < self.chain_view.burn_block_height {
+                (target_block_height + (BLOCKS_INV_DATA_MAX_BITLEN as u64), BLOCKS_INV_DATA_MAX_BITLEN as u64)
+            }
+            else {
+                (self.chain_view.burn_block_height, self.chain_view.burn_block_height - target_block_height)
```

Comment:
Can `taget_block_height` ever be greater than chain_view.burn_block_height?

Reply:
It shouldn't.  Looking at all the places where `make_highest_getblocksdata` gets called, it should never be the case that `target_block_height` is greater than `self.chain_view.burn_block_height`.  But I can add a defensive check.

Reply:
Fixed in 82179fb46